### DISCUSSION TITLE
I used socat to enable external programs to access the Zigbee interface of S6E.

### DISCUSSION THREAD

**Discussion by maodun96** [seaky/nspanel_pro_tools_apk#147] [Ideas]
I used socat to enable external programs to access the Zigbee interface of S6E. This could potentially serve as a built-in feature of the project.

**Steps:**
1. Download socat from [here](https://github.com/jakev/android-binaries/blob/master/socat).
   
2. Run the following commands:
   ```bash
   su
   mount -o remount,rw /system
   cp /storage/emulated/0/socat /system/bin/socat

   chmod +x /system/bin/socat
   export PATH=$PATH:/system/bin
   mount -o remount,ro /system
   ```

3. Open Auto Magic and set it to start "Execute Root Command" at boot:
   ```bash
   /system/bin/socat -d -d -d -d tcp-listen:8889,reuseaddr,fork file:/dev/ttyS5,b115200,raw,echo=0
   ```

At this point, the port `tcp://device's IP:8889` can be used by zigbee2mqtt or ZHA.

**Comment by maodun96** [comment]
The method comes from Blakadder's X: https://x.com/blakadder_/status/1574490845818109964?s=20.

**Comment by maodun96** [comment]
I also ran Z2M on S6E through Termux. It seemed to put a certain load on the device. So I eventually switched to using HA's Z2M and called the relevant devices from S6E.

**Comment by seaky** [comment]
very neat, congrat!
